feat: add application feature

murich opened the pull request:
## Summary

This PR introduces comprehensive application management functionality to the Casdoor CLI, enabling developers to perform full CRUD operations on OIDC applications programmatically.

## Features Added

### New Commands
- `casdoor applications list` - List all applications with detailed information
- `casdoor applications add` - Create new OIDC applications with full configuration
- `casdoor applications delete` - Remove applications by name

### Key Capabilities
- **Complete OIDC Support**: All standard OAuth2/OIDC parameters including client credentials, redirect URIs, grant types, and scopes
- **SDK Integration**: Built on official Casdoor Go SDK for reliable API communication
- **Table Formatting**: Clean, readable output for application listings
- **Comprehensive Validation**: Required field validation and user-friendly error messages
- **Flexible Configuration**: Support for multiple redirect URIs, custom descriptions, and organization assignment

### Technical Implementation
- Added `Application` model with complete field mapping
- Created `ApplicationManager` helper for centralized operations
- Implemented proper error handling and user feedback
- Fixed command flag conflicts and ensured clean CLI structure
- Added utility functions for table formatting

## Usage Examples

```bash
# List all applications
casdoor applications list

# Create a new application
casdoor applications add \
  --name "MyApp" \
  --client-id "my-app-client" \
  --client-secret "secure-secret" \
  --redirect-uris "http://localhost:3000/callback" \
  --description "My application"

# Delete an application
casdoor applications delete --name "MyApp"
```

## Testing

Comprehensive testing performed against live Casdoor instance:
- ✅ List operations showing all existing applications
- ✅ Add operations creating applications with proper OIDC configuration
- ✅ Delete operations removing applications cleanly
- ✅ Error handling for invalid inputs and permissions
- ✅ Table formatting displaying all relevant application details

## Benefits

- **Developer Productivity**: No need to use web interface for application management
- **Automation**: Enables scripting and CI/CD integration
- **Consistency**: Standardized command structure following existing CLI patterns
- **Reliability**: Built on official SDK with proper error handling

This enhancement significantly improves the developer experience for managing Casdoor applications and enables better integration into automated workflows.

murichGhostdrive committed:
feat: add comprehensive application management functionality

- Add applications command group with list, add, delete subcommands
- Implement full CRUD operations for Casdoor applications via SDK
- Add Application model with complete OIDC configuration support
- Create ApplicationManager helper for centralized app operations
- Add proper error handling and user-friendly feedback
- Include table formatting for application listing
- Fix flag conflicts and ensure clean command structure
- Support all standard OIDC parameters and OAuth2 flows

This enhancement enables complete application lifecycle management
through the CLI, providing developers with programmatic access to
Casdoor's application configuration without requiring web interface.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/casdoor/casdoor-cli?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/casdoor/casdoor-cli?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/casdoor/casdoor-cli?pullRequest=1) it.</sub>

murichGhostdrive committed:
fix: restore Go 1.22.0 requirement and verify compatibility

- Upgrade go.mod back to Go 1.22.0 from unnecessarily downgraded 1.21
- Verify all functionality works correctly with Go 1.22.0
- Rebuild binary with updated Go version requirement
- All tests continue to pass with higher Go version

The downgrade to 1.21 was unnecessary as we're building with Go 1.24
which supports both versions. Go 1.22.0 is more appropriate for
modern development and aligns with current Go best practices.

hsluoyz commented:
@murich sign CLA:

![image](https://github.com/user-attachments/assets/78bb74f9-e6c2-4a38-b643-3daad470fc55)
